azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -295,31 +314,36 @@ public void cleanUpDanglingYarnApplications() {
         ii. filter and get the yarn applications that within the union set of above executions
         iii. kill these yarn applications
      */
-    logger.info("cleanUpDanglingYarnApplications start ");
+    try {
+      logger.info("cleanUpDanglingYarnApplications start ");
 
-    Set<Integer> executionStoppedFlows = getExecutionStoppedFlows();
-    logger.info("Get executionStoppedFlows: " +
-        executionStoppedFlows.stream().map(Object::toString).collect(Collectors.joining(",")));
+      Set<Integer> executionStoppedFlows = getExecutionStoppedFlows();
+      logger.info("Get executionStoppedFlows: " +
+          executionStoppedFlows.stream().map(Object::toString).collect(Collectors.joining(",")));
 
-    // get those flows terminated but containers are still alive (failed to properly killed)
-    Set<Integer> toBeCleanedContainers = getContainersOfTerminatedFlows();
-    logger.info("Get terminatedContainers: " +
-        toBeCleanedContainers.stream().map(Object::toString).collect(Collectors.joining(",")));
+      // get those flows terminated but containers are still alive (failed to properly killed)
+      Set<Integer> toBeCleanedContainers = getContainersOfTerminatedFlows();
+      logger.info("Get terminatedContainers: " +
+          toBeCleanedContainers.stream().map(Object::toString).collect(Collectors.joining(",")));
 
-    // combine both sets
-    toBeCleanedContainers.addAll(executionStoppedFlows);
-    logger.info("Get the set of all executions: " +
-        toBeCleanedContainers.stream().map(Object::toString).collect(Collectors.joining(",")));
+      // combine both sets
+      toBeCleanedContainers.addAll(executionStoppedFlows);
+      logger.info("Get the set of all executions: " +
+          toBeCleanedContainers.stream().map(Object::toString).collect(Collectors.joining(",")));
 
-    if (toBeCleanedContainers.isEmpty()) {
-      logger.info("No execution needs to kill yarn application, exit");
-      return;
-    }
+      if (toBeCleanedContainers.isEmpty()) {
+        logger.info("No execution needs to kill yarn application, exit");
+        return;
+      }
 
-    // For each of yarn clusters: find applications of the above executionIDs and kill them
-    for (Entry<String, Cluster> entry : this.allClusters.entrySet()) {
-      logger.info("clean up yarn applications in cluster:" + entry.getValue().clusterId);
-      cleanUpYarnApplicationsInCluster(toBeCleanedContainers, entry.getValue());
+      // For each of yarn clusters: find applications of the above executionIDs and kill them
+      for (Entry<String, Cluster> entry : this.allClusters.entrySet()) {
+        logger.info("clean up yarn applications in cluster:" + entry.getValue().clusterId);
+        cleanUpYarnApplicationsInCluster(toBeCleanedContainers, entry.getValue());
+      }
+    } catch (Throwable t) {
```

Comment:
If we do not add catch here, what will happen? Will it terminate the start() function immediately?

Reply:
it will fail this call and will no longer schedule any of the future call of `scheduleAtFixedRate` method